Config/Config.php

```diff
@@ -0,0 +1,60 @@
+<?php
+namespace mvrhov\PhinxBundle\Config;
+
+use Phinx\Config\Config as PhinxConfig;
+
+class Config extends PhinxConfig {
+
+    public function __construct(array $configArray, $configFilePath = null)
+    {
+        if (array_key_exists('environments', $configArray) && is_array($configArray['environments'])) {
+            $configArray['environments'] = $this->preprocessEnvironmentsConfiguration($configArray['environments']);
+        }
+
+        return parent::__construct($configArray, $configFilePath);
+    }
+
+    private function preprocessEnvironmentsConfiguration(array $environments)
+    {
+        foreach ($environments as $environment => $environmentData) {
+            if (!is_array($environmentData)) {
+                //This is some key like: ("default_database" => "default"),
+                //not an actual environment definition.
+                continue;
+            }
+
+            if (!array_key_exists('url', $environmentData)) {
+                //This environment likely contains (host, name, user, pass, port) like Phinx expects.
+                //No need to try and parse those out of a 'url' config value.
+                continue;
+            }
+
+            $matches = null;
+            if (!preg_match('/^([^:]+):\/\/([^:]+):([^@]+)@([^\/]+)\/(.+?)$/', $environmentData['url'], $matches)) {
```

Comment:
I'm not fond of parsing this with the regex.
I'd rather see that you use the same logic as [DBAL](https://github.com/doctrine/dbal/blob/d9aaf5429ac77e9daf8ef0c06260bb0f3bcc189b/lib/Doctrine/DBAL/DriverManager.php#L236) does